addons/l10n_it_stock_ddt/tests/test_edi.py

```diff
@@ -0,0 +1,151 @@
+                'categ_id': product_category.id,
+                'standard_price': 55.0,
+                'list_price': 70.0,
+                'type': 'consu',
+                'weight': 0.01,
+                'uom_id': uom_unit.id,
+                'uom_po_id': uom_unit.id,
+                'default_code': 'FURN_7777',
+                'invoice_policy': 'delivery',
+                'expense_policy': 'no',
+                'taxes_id': [(6, 0, [])],
+                'supplier_taxes_id': [(6, 0, [])],
+            }
+        ])
+
+    def test_deferred_invoice(self):
+        """ Create a sale order with multiple DDTs, and create an invoice with a later date.
+            The export has to have the TipoDocumento TD24 for Deferred Invoice.
+        """
+        # Create sale order
+        with freeze_time('2020-02-02 18:00'):
+            self.sale_order = self.env['sale.order'].with_company(self.company).create({
+                'partner_id': self.italian_partner_a.id,
+                'partner_invoice_id': self.italian_partner_a.id,
+                'partner_shipping_id': self.italian_partner_a.id,
+                'order_line': [
+                    (0, 0, {
+                        'name': product.name,
+                        'product_id': product.id,
+                        'product_uom_qty': 5,
+                        'product_uom': product.uom_id.id,
+                        'price_unit': product.list_price,
+                        'tax_id': self.tax_22
+                    }) for product in self.products
+                ],
+                'pricelist_id': self.default_pricelist.id,
+                'picking_policy': 'direct',
+            })
+            self.sale_order.action_confirm()
+
+            # Create two pickings, so 2 DDTs
+            for dummy in range(2):
+                self._create_delivery(self.sale_order, 1)
+
+        # Create one invoice
+        with freeze_time('2020-02-03 09:00'):
+            deferred_invoice = self.sale_order._create_invoices()
+            deferred_invoice.action_post()
+
+        # Check the XML output of the invoice
+        self._test_invoice_with_sample_file(
+            deferred_invoice,
+            "deferred_invoice.xml",
+            xpaths_result={"//DatiGeneraliDocumento/Numero": "<Numero/>",},
+            xpaths_file={"//DatiGeneraliDocumento/Numero": "<Numero/>",}
+        )
+
+    def _create_delivery(self, sale_order, qty=1):
+        """ Create a picking of a limited quantity and create a backorder """
+        pickings = sale_order.picking_ids.filtered(lambda picking: picking.state != 'done')
+        pickings.move_ids.write({'quantity_done': qty})
+        wizard_action = pickings.button_validate()
+        context = wizard_action['context']
+        wizard = Form(self.env['stock.backorder.confirmation'].with_context(context))
+        confirm_dialog = wizard.save()
+        confirm_dialog.process()
+
+    @classmethod
+    def _get_test_file_content(cls, filename):
```

Comment:
Overridden function, the path changes and I didn't want to add a "path" parameter to a function that opens a file or security, even if it's in a test